### PR TITLE
Add issues new

### PR DESCRIPTION
**Pull request by PinkhamSoftware**
What:
- Issue Types Dropdown list
- Issue Locations Dropdown list
- Issue Locations Gateway (json) with promise

Why:
- Swap out json file for api call quite easily
- Service container injection for resolving gateway

Notes:
- Dropdown list does requires unique key stored in the object to be able to change value properly and distinguish one element from the other.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,8 +52,10 @@
     "enzyme": "^3.9.0",
     "enzyme-adapter-react-16": "^1.13.2",
     "jest-localstorage-mock": "^2.4.0",
+    "json-loader": "^0.5.7",
     "react-docgen-typescript-webpack-plugin": "^1.1.0",
     "testcafe": "^1.2.0",
-    "testcafe-react-selectors": "^3.1.0"
+    "testcafe-react-selectors": "^3.1.0",
+    "ts-loader": "^6.0.2"
```
Comment (danielwilliams94) [inline package.json]: Why have we added ts-loader?

Comment (PinkhamSoftware) [reply package.json]: I think that was an attempt at an approach to loading json, not sure we need it now.

File: src/Boundary/IssueLocation/index.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+export interface IIssueLocationGateway {
+    loadIssueLocations: () => Promise<ILoadIssuesOutputMeetingModel>;
+}
+
+export interface ILoadIssuesOutputMeetingModel{
+    successful: boolean,
+    issueLocations: IIssueLocation[]
+}
+
+export class LoadIssuesOutputMeetingModel implements ILoadIssuesOutputMeetingModel{
```
Comment (danielwilliams94) [inline src/Boundary/IssueLocation/index.ts]: Typically we don't put the actual implementation of the interfaces in the boundary, only the interfaces themselves. The class implementation should be alongside the use case (I don't think you have one) - the gateway shouldn't be returning this model, it should just be returning the data from the request.

File: src/Components/AddIssue/index.tsx [hunk 3]
```diff
@@ -1,71 +1,139 @@
...
-              [name]: [value]
-          }
-      });
+
+      let issue = this.state.issue;
+      issue.Notes = value;
+      this.setState({issue:issue});
+
+      this.props.onChangeIssue(this.state.issue, this.props.index);
+      console.log(this.state.issue);
```
Comment (danielwilliams94) [inline src/Components/AddIssue/index.tsx]: Probably remove all the console.log statements

File: src/Components/AddIssue/index.tsx [hunk 4]
```diff
@@ -1,71 +1,139 @@
...
+      this.setState({issue:issue });
 
-      this.props.onChangeIssue(this.state.issue, this.props.index)
+      this.props.onChangeIssue(this.state.issue, this.props.index);
+      console.log(this.state.issue);
     }
 
+    createSelectItemsForIssueTypes() {
+      let items = [];         
+      for (let i = 0; i < this.state.issueTypes.length ; i++) {
```
Comment (danielwilliams94) [inline src/Components/AddIssue/index.tsx]: I really think these would be cleaner as map statements but this works so not a big deal

File: src/Domain/IssueType/index.ts [hunk 5]
```diff
@@ -1,14 +1,19 @@
...
-   IssueId:string
-   IssueType:string
+   //key:any;
+   IssueId:string;
+   IssueType:string;
 } 
 
 export class IssueType implements IIssueType
 {
+   //public key:any;
```
Comment (danielwilliams94) [inline src/Domain/IssueType/index.ts]: I think you need to uncomment these right?

File: src/Domain/IssueType/index.ts [hunk 6]
```diff
@@ -1,14 +1,19 @@
...
    public IssueId:string;
    public IssueType:string;
    public constructor(issueType:string, issueId:string){
+      //this.key = uuid();
```
Comment (danielwilliams94) [inline src/Domain/IssueType/index.ts]: This one too?

File: src/Boundary/IssueLocation/index.ts (hunk 2 above)
Comment (PinkhamSoftware) [reply src/Boundary/IssueLocation/index.ts]: Ok, I'll move this into the UseCase folder.

File: src/Components/AddIssue/index.tsx (hunk 4 above)
Comment (Rumaysa) [reply src/Components/AddIssue/index.tsx]: Ok, will change that

File: src/Domain/IssueType/index.ts (hunk 5 above)
Comment (Rumaysa) [reply src/Domain/IssueType/index.ts]: Yeah, will update that.

File: src/Domain/IssueType/index.ts (hunk 6 above)
Comment (Rumaysa) [reply src/Domain/IssueType/index.ts]: Yeah, will also update this.

File: src/Components/AddIssue/index.tsx (hunk 3 above)
Comment (PinkhamSoftware) [reply src/Components/AddIssue/index.tsx]: Yep agreed.

**Comment by PinkhamSoftware** [comment]
Yep thanks going to address them 👍

File: src/Domain/IssueType/index.ts (hunk 6 above)
Comment (Rumaysa) [reply src/Domain/IssueType/index.ts]: Resolved in https://github.com/LBHackney-IT/lbh-etra-front-end/pull/14/commits/cd35796cfbf5d352ac16f566ab172f842edf4add

File: src/Domain/IssueType/index.ts (hunk 5 above)
Comment (Rumaysa) [reply src/Domain/IssueType/index.ts]: Resolved in https://github.com/LBHackney-IT/lbh-etra-front-end/pull/14/commits/cd35796cfbf5d352ac16f566ab172f842edf4add.

File: src/Components/AddIssue/index.tsx (hunk 4 above)
Comment (Rumaysa) [reply src/Components/AddIssue/index.tsx]: Resolved in https://github.com/LBHackney-IT/lbh-etra-front-end/pull/14/commits/7eae314f2f10a1808c5446a1e12922d269af4536

File: src/Components/AddIssue/index.tsx (hunk 3 above)
Comment (Rumaysa) [reply src/Components/AddIssue/index.tsx]: Resolved in 
49ae60f

File: src/Boundary/IssueLocation/index.ts (hunk 2 above)
Comment (Rumaysa) [reply src/Boundary/IssueLocation/index.ts]: Resolved in https://github.com/LBHackney-IT/lbh-etra-front-end/pull/14/commits/263b6ebeef3305a9d94fb064a9a5564c6f7945a1

File: package.json (hunk 1 above)
Comment (Rumaysa) [reply package.json]: Resolved in https://github.com/LBHackney-IT/lbh-etra-front-end/pull/14/commits/b09e38d4f593f367389bfc8dbaedd7dd393c6743.

**Pull request merged**